File: lua/bob/bob.lua
local bob = {}

local QUESTION_RESPONSE = 'Sure'
local SILENCE_RESPONSE = 'Fine, be that way.'
local YELLING_RESPONSE = 'Whoa, chill out!'
local DEFAULT_RESPONSE = 'Whatever'

local function isQuestion(talk)
  return talk:sub(talk:len()) == '?'
end

local function isYelling(talk)
  return talk:upper() == talk
end

local function isSilence(talk)
  return talk:len() == 0
end

function bob.hey(talk)
  local response
  if (isQuestion(talk)) then response = QUESTION_RESPONSE
  elseif (isSilence(talk)) then response = SILENCE_RESPONSE
  elseif (isYelling(talk)) then response = YELLING_RESPONSE
  else response = DEFAULT_RESPONSE end
  return response
end

return bob

File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    private static string DEFAULT_RESPONSE = "Whatever.";
    private static string QUESTION_RESPONSE = "Sure.";
    private static string SILENCE_RESPONSE = "Fine. Be that way!";
    private static string YELLING_RESPONSE = "Whoa, chill out!";

    public static string Response(string statement)
    {
        string response = DEFAULT_RESPONSE;
        if (IsQuestion(statement)) {
          response = QUESTION_RESPONSE;
        }
        else if (IsSilence(statement)) {
          response = SILENCE_RESPONSE;
        }
        else if (IsYelling(statement)) {
          response = YELLING_RESPONSE;
        }
        return response;
    }

    private static bool IsQuestion(string statement)
    {
      return statement.EndsWith("?");
    }

    private static bool IsSilence(string statement)
    {
      return false;
    }

    private static bool IsYelling(string statement)
    {
      return statement.ToUpper() == statement;
    }
}
